lib/elixir/lib/calendar/datetime.ex

```diff
@@ -519,6 +519,12 @@ defmodule DateTime do
       {hour, minute, second} = unquote(read_time)
 
       cond do
+        offset == 0 ->
```

Comment:
Actually, I think this will be even faster if you move this clause [here](https://github.com/dcuddeback/elixir/blob/46f057f5546f45f254176701cfa5619bed46e963/lib/elixir/lib/calendar/datetime.ex#L536). And then you never call NaiveDateTime, you just build the DateTime directly, [like done here](https://github.com/dcuddeback/elixir/blob/46f057f5546f45f254176701cfa5619bed46e963/lib/elixir/lib/calendar/datetime.ex#L205-L218).

Reply:
You were a few seconds faster than me 😂 

I suggested to put it before `is_nil(offset)` to speed up the happy case, what do you think?

Reply:
Sounds good to me too!

Reply:
Makes sense. We won't be using the change that I made to `NaiveDateTime.new/7` anymore: https://github.com/elixir-lang/elixir/pull/7986/commits/46f057f5546f45f254176701cfa5619bed46e963. Still keep that? Or should I pull that out when I make this change?